services/QuillLMS/spec/workers/set_impact_metrics_worker_spec.rb

```diff
@@ -5,54 +5,56 @@
 describe SetImpactMetricsWorker do
   subject { described_class.new }
 
-  describe '#perform' do
-    let!(:school) { create(:school) }
-    let!(:low_income_school) { create(:school, free_lunches: 40) }
-    let!(:first_classroom) { create(:classroom_with_students_and_activities) }
-    let!(:second_classroom) { create(:classroom_with_students_and_activities) }
-    let!(:first_schools_user) { create(:schools_users, school: school, user: first_classroom.owner)}
-    let!(:second_schools_user) { create(:schools_users, school: low_income_school, user: second_classroom.owner)}
-    let!(:archived_activity_session) { create(:activity_session, user: User.where(role: 'student').first, visible: false )}
-
-    before(:all) do
-      @number_of_sentences = ActivitySession.unscoped.where(state: 'finished').count.floor(-5) * 10
-      @number_of_students = ActivitySession.unscoped.where(state: 'finished').count('DISTINCT(user_id)')
-      teachers_query = User.select(:id)
-        .joins(:units, classroom_units: :activity_sessions)
-        .where("activity_sessions.state = 'finished'")
-        .group('users.id')
-        .having('count(activity_sessions) > ?', SetImpactMetricsWorker::ACTIVITY_SESSION_MINIMUM)
-      teacher_ids = teachers_query.to_a.map(&:id)
-      @number_of_teachers = teachers_query.to_a.count.floor(-2)
-      schools_query = School.joins(:schools_users).where(schools_users: {user_id: teacher_ids}).distinct
-      @number_of_schools = schools_query.to_a.count
-      low_income_schools_query = schools_query.where("free_lunches > ?", SetImpactMetricsWorker::FREE_LUNCH_MINIMUM)
-      @number_of_low_income_schools = low_income_schools_query.to_a.count
+  context '#perform' do
+    let(:activity_sessions) { create_list(:activity_session, 100)}
+
+    let(:example_json1) {
+      activity_sessions.map {|as| {"id" => as.id}}
+    }
+
+    let(:teachers) { create_list(:teacher, 150)}
+
+    let(:example_json2) {
+      teachers.map {|t| {"id" => t.id}}
+    }
+
+    let(:example_json3) {
+      [
+        {"id" => 333, "free_lunches" => 30},
+        {"id" => 334, "free_lunches" => 0},
+        {"id" => 545, "free_lunches" => 54}
+      ]
+    }
+
+    before do
+      allow(QuillBigQuery::ActivitiesAllTimeQuery).to receive(:run).and_return(example_json1)
+      allow(QuillBigQuery::ActiveTeachersAllTimeQuery).to receive(:run).and_return(example_json2)
+      allow(QuillBigQuery::SchoolsContainingCertainTeachersQuery).to receive(:run).with(teachers.map(&:id)).and_return(example_json3)
```

Comment:
Super minor, but you can replace `map(&:id)` with `pluck(:id)` which I THINK is a pattern that we're preferring when we can get away with it?